### PR TITLE
Create Archive Goal Modal

### PR DESCRIPTION
**Pull request by 7malikk**
When Archive Goal button in the navbar is clicked Archive Goal Modal pops up
Solves issue #247

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,15 +16,16 @@
     "author": "Team Plato",
     "license": "MIT",
     "dependencies": {
+        "@reduxjs/toolkit": "^1.6.1",
         "axios": "^0.21.1",
-        "joi": "^17.4.2",
```
Comment (abiona01) [inline package.json:20]: Why did you remove this?

File: client/src/components/navbar/Navbar.jsx [hunk 2]
```diff
@@ -52,6 +139,92 @@ const Navbar = () => {
...
+
+// const SettingsIconStyled = styled(SettingsIcon)`
+//   color: #999999;
+//   margin-left: 15px;
+//   cursor: pointer;
+// `;
+// const Header = styled.h2`
+//   background: #f6f6f6;
+//   padding: 0.1rem;
+//   margin-top: -10px;
```
Comment (abiona01) [inline client/src/components/navbar/Navbar.jsx:226]: Why did you add this block of code only to comment it out?

File: client/src/components/navbar/Navbar.jsx [hunk 3]
```diff
@@ -37,7 +77,54 @@ const Navbar = () => {
...
+                    Clicking the proceed button means that people will no longer have access to view this goal.
+                  </p>
+                  <div
+                    style={{
+                      display: 'flex',
+                      alignContent: 'center',
+                      justifyContent: 'center',
+                      textAlign: 'center',
+                      marginTop: '1rem',
+                      paddingLeft: '1.1rem',
```
Comment (abiona01) [inline client/src/components/navbar/Navbar.jsx:119]: Styled components for styling. These styles are too much for you to add as inline. Please review

File: client/src/components/navbar/Navbar.jsx [hunk 4]
```diff
@@ -1,3 +1,14 @@
+import React from 'react';
+
+import IconButton from '@material-ui/core/IconButton';
+import Modal from '@material-ui/core/Modal';
+import { makeStyles } from '@material-ui/core/styles';
+import AddIcon from '@material-ui/icons/Add';
+import ClearIcon from '@material-ui/icons/Clear';
+import SearchIcon from '@material-ui/icons/Search';
+import SettingsIcon from '@material-ui/icons/Settings';
+import styled from 'styled-components';
```
Comment (abiona01) [inline client/src/components/navbar/Navbar.jsx:10]: The Navbar component should house all other components displayed in it. You should not build your component in the navbar component. Please review

File: client/src/components/navbar/Navbar.jsx [hunk 5]
```diff
@@ -18,14 +29,43 @@ import {
...
+    padding: theme.spacing(2, 4, 3),
+    top: '15%',
+    left: '33%',
+  },
+}));
+
 const Navbar = () => {
+  const classes = useStyles();
+
+  const [open, setOpen] = React.useState(false);
```
Comment (abiona01) [inline client/src/components/navbar/Navbar.jsx:53]: useState has been imported at the top of the code. Importing it here is redundant. Please review

File: client/src/components/navbar/Navbar.jsx [hunk 6]
```diff
@@ -37,7 +77,54 @@ const Navbar = () => {
...
+                  </div>
+                  <h2
+                    style={{
+                      display: 'flex',
+                      alignContent: 'center',
+                      justifyContent: 'center',
+                      textAlign: 'center',
+                      margin: '1rem 0',
+                      color: '#00B87C',
+                    }}
```
Comment (abiona01) [inline client/src/components/navbar/Navbar.jsx:105]: Style this component with styled component.

File: client/src/components/navbar/Navbar.jsx (hunk 3 above)
Comment (abiona01) [inline client/src/components/navbar/Navbar.jsx:119]: Style with styled components

**Review by abiona01 — CHANGES_REQUESTED** [review]

**Comment by KelvinAkpobome** [comment]
You are making PR from a stale branch

**Comment by 7malikk** [comment]
> You are making PR from a stale branch

What's the current branch? Everyone is staying staging, which I pulled from in the first place...

**Comment by abiona01** [comment]
You don't have conflicts anymore. Just work on the lines I highlighted.
